Subject: [PATCH] rild: Allow RIL_Init to be run before switching to radio user

Some older blobs need this to properly setup their modem.

diff --git a/rild/rild.c b/rild/rild.c
index b32afe4..f089c5e 100644
--- a/rild/rild.c
+++ b/rild/rild.c
@@ -328,7 +328,6 @@ int main(int argc, char **argv) {
     }
 OpenLib:
 #endif
-    switchUser();
 
     dlHandle = dlopen(rilLibPath, RTLD_NOW);
 
@@ -406,6 +405,8 @@ OpenLib:
 
     RLOGD("RIL_register_socket completed");
 
+    switchUser();
+
 done:
 
     RLOGD("RIL_Init starting sleep loop");
